### ISSUE TITLE
[Bug]: It takes more than 5 minutes to load into memory

### ISSUE THREAD

**Issue by zyy20191** [milvus-io/milvus#21159]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:2.2
- Deployment mode(standalone or cluster):
- SDK version(e.g. pymilvus v2.0.0rc2):pymilvus v2.2
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1.I've set up an index
2.The load code used is:
        collection.load(partition_names=[xxx,xxx])

3.I created five partitions, three of which had zero data.I don't know if it will affect

### Milvus Log

[2022/12/12 09:55:29.535 +00:00] [DEBUG] [proxy/impl.go:829] ["GetCollectionStatistics received"] [traceID=489f0c53fd449c57] [role=proxy] [db=] [collection=xiaowei_platform]
[2022/12/12 09:55:29.535 +00:00] [DEBUG] [proxy/impl.go:856] ["GetCollectionStatistics enqueued"] [traceID=489f0c53fd449c57] [role=proxy] [msgID=438000400339304456] [BeginTS=438000400339304456] [EndTS=438000400339304456] [db=] [collection=xiaowei_platform]
[2022/12/12 09:55:29.535 +00:00] [INFO] [datacoord/services.go:282] ["received request to get collection statistics"] [traceID=489f0c53fd449c57] [module=DataCoord]
[2022/12/12 09:55:29.535 +00:00] [INFO] [datacoord/services.go:295] ["success to get collection statistics"] [traceID=489f0c53fd449c57] [module=DataCoord] [response="stats:<key:\"row_count\" value:\"142011\" > status:<> "]
[2022/12/12 09:55:29.536 +00:00] [DEBUG] [proxy/impl.go:889] ["GetCollectionStatistics done"] [traceID=489f0c53fd449c57] [role=proxy] [msgID=438000400339304456] [BeginTS=438000400339304456] [EndTS=438000400339304456] [db=] [collection=xiaowei_platform]
[2022/12/12 09:55:29.536 +00:00] [DEBUG] [proxy/impl.go:2376] ["GetIndexState received"] [traceID=202b0335e2bf1b97] [role=proxy] [db=] [collection=xiaowei_platform] [field=] ["index name"=]
[2022/12/12 09:55:29.537 +00:00] [DEBUG] [proxy/impl.go:2407] ["GetIndexState enqueued"] [traceID=202b0335e2bf1b97] [role=proxy] [MsgID=438000400339304457] [BeginTs=438000400339304457] [EndTs=438000400339304457] [db=] [collection=xiaowei_platform] [field=] ["index name"=]
[2022/12/12 09:55:29.537 +00:00] [INFO] [indexcoord/index_coord.go:490] ["IndexCoord get index state"] [collectionID=437728473907200070] [indexName=_default_idx]
[2022/12/12 09:55:29.537 +00:00] [ERROR] [indexcoord/index_coord.go:506] ["IndexCoord get index state fail"] [collectionID=437728473907200070] [indexName=_default_idx] ["fail reason"="there is no index on collection: 437728473907200070 with the index name: _default_idx"] [stack="github.com/milvus-io/milvus/internal/indexcoord.(*IndexCoord).GetIndexState\n\t/go/src/github.com/milvus-io/milvus/internal/indexcoord/index_coord.go:506\ngithub.com/milvus-io/milvus/internal/distributed/indexcoord.(*Server).GetIndexState\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/indexcoord/service.go:256\ngithub.com/milvus-io/milvus/internal/proto/indexpb._IndexCoord_GetIndexState_Handler.func1\n\t/go/src/github.com/milvus-io/milvus/internal/proto/indexpb/index_coord.pb.go:2448\ngithub.com/milvus-io/milvus/internal/util/logutil.UnaryTraceLoggerInterceptor\n\t/go/src/github.com/milvus-io/milvus/internal/util/logutil/grpc_interceptor.go:22\ngithub.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1\n\t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25\ngithub.com/grpc-ecosystem/go-grpc-middleware/tracing/opentracing.UnaryServerInterceptor.func1\n\t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/tracing/opentracing/server_interceptors.go:38\ngithub.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1\n\t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25\ngithub.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1\n\t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:34\ngithub.com/milvus-io/milvus/internal/proto/indexpb._IndexCoord_GetIndexState_Handler\n\t/go/src/github.com/milvus-io/milvus/internal/proto/indexpb/index_coord.pb.go:2450\ngoogle.golang.org/grpc.(*Server).processUnaryRPC\n\t/go/pkg/mod/google.golang.org/grpc@v1.46.0/server.go:1283\ngoogle.golang.org/grpc.(*Server).handleStream\n\t/go/pkg/mod/google.golang.org/grpc@v1.46.0/server.go:1620\ngoogle.golang.org/grpc.(*Server).serveStreams.func1.2\n\t/go/pkg/mod/google.golang.org/grpc@v1.46.0/server.go:922"]
[2022/12/12 09:55:29.538 +00:00] [DEBUG] [proxy/impl.go:2443] ["GetIndexState done"] [traceID=202b0335e2bf1b97] [role=proxy] [MsgID=438000400339304457] [BeginTs=438000400339304457] [EndTs=438000400339304457] [db=] [collection=xiaowei_platform] [field=] ["index name"=_default_idx]

**Comment by yanliang567** [comment]
/assign @binbinlv 
@binbinlv can we reproduce this locally

/unassign

**Comment by yanliang567** [comment]
@zyy20191 Could you please refer this [script](https://github.com/milvus-io/milvus/tree/master/deployments/export-log) to export the whole Milvus logs for investigation?

**Comment by xiaofan-luan** [comment]
@xiaocai2333 could you verify this is the same issue you fix?

**Comment by zyy20191** [comment]
I just loaded the collection into memory again and exported the logs.
[milvus.log](https://github.com/milvus-io/milvus/files/10213594/milvus.log)
The logs file is too big to upload here, so I took a portion of it this morning. I put the full file here:

**Comment by zyy20191** [comment]
@yanliang567 Here is the complete log file.

链接：https://pan.baidu.com/s/1wyhtk73LDrWyJsvUvMnXag 
提取码：afh2

**Comment by zyy20191** [comment]
@xiaofan-luan By the way, my other server was also slow to load, but did not report this error: "IndexCoord get index state fail", so there may be other factors causing the problem.

**Comment by xiaofan-luan** [comment]
/assign @xiaocai2333 
please help on investigating, thanks

**Comment by zyy20191** [comment]
@xiaocai2333 Except for the slow loading speed, no errors have been found in other services and the program can run normally.

**Comment by zyy20191** [comment]
@xiaocai2333 
I noticed that my newly created collection is loading more slowly than it did yesterday.It seems that the longer it takes to create, the longer it takes to load.
I did not do any operation on the collection during this period.

**Comment by xiaocai2333** [comment]
> 

ok, thanks for  reporting this issue, I will debug it.

**Comment by binbinlv** [comment]
> /assign @binbinlv @binbinlv can we reproduce this locally
> 
> /unassign

Did not reproduce this issue on local side, @zyy20191 Could you please have a look whether the following script represents your scenario? Thanks.

milvus: standalone & cluster
milvus version: 2.2.0
pymilvus version: 2.2.0
result: Load costs 3.031479 s

script:
```python
from pymilvus import CollectionSchema, FieldSchema
from pymilvus import Collection
from pymilvus import connections
from pymilvus import DataType
from pymilvus import Partition
from pymilvus import utility
import time

dim=768
int64_field = FieldSchema(name="int64", dtype=DataType.INT64, is_primary=True)
int64_field_1 = FieldSchema(name="int641", dtype=DataType.INT64)
int64_field_2 = FieldSchema(name="int642", dtype=DataType.INT64)
int64_field_3 = FieldSchema(name="int643", dtype=DataType.INT64)
int64_field_4 = FieldSchema(name="int644", dtype=DataType.INT64)
float_vector = FieldSchema(name="float_vector", dtype=DataType.FLOAT_VECTOR, dim=dim)
schema = CollectionSchema(fields=[int64_field, int64_field_1, int64_field_2, int64_field_3, int64_field_4, float_vector], auto_id=True)

connections.connect(host="10.100.31.107", port="19530")
import numpy as np
collection = Collection("test_21159", schema=schema)
collection.create_partition(partition_name="search_partition_1", description="search partition 1")
collection.create_partition(partition_name="search_partition_2", description="search partition 2")
collection.create_partition(partition_name="search_partition_3", description="search partition 3")
collection.create_partition(partition_name="search_partition_4", description="search partition 4")

partitions = collection.partitions
nb = 50000
import random
vectors = [[random.random() for _ in range(dim)] for _ in range(nb)]
collection.insert([[i for i in range(nb)], [i for i in range(nb)], [i for i in range(nb)], [i for i in range(nb)], vectors],  partitions[0].name)
vectors = [[random.random() for _ in range(dim)] for _ in range(nb)]
collection.insert([[i for i in range(nb)], [i for i in range(nb)], [i for i in range(nb)], [i for i in range(nb)], vectors],  partitions[1].name)
index_param = {"index_type": "IVF_SQ8", "metric_type": "L2", "params": {"nlist": 1024}}
collection.create_index("float_vector", index_param, index_name="index_name_1")
start_time = time.time()
collection.load(partition_names=[partitions[0].name, partitions[1].name])
end_time = time.time()
print("Load costs %f s" % (end_time-start_time))
```

**Comment by zyy20191** [comment]
**index_type: IVF_FLAT**

I was also quick to load this way. But after a few days of use, when I release the collection and use the load method alone, the load becomes slow.

Assuming the server is down, reload the collection into memory.So I just did the load operation and found the problem.

**Comment by zyy20191** [comment]
@binbinlv My scenario is that the collection has been used normally for a while, and then release it -> load it. In this case, the load is slow.

**Comment by binbinlv** [comment]
> @binbinlv My scenario is that the collection has been used normally for a while, and then release it -> load it. In this case, the load is slow.

@zyy20191 OK, got it, thanks for your explanation.

**Comment by zyy20191** [comment]
@binbinlv I created the collection at 2022-12-12 18:45:30. Just now, after I released it and then reloaded it. The loading speed was 1 minute. Nothing is done to it during this period.

So the loading speed does get slower and slower over time.

**Comment by yanliang567** [comment]
@binbinlv is trying to reproduce the issue locally

**Comment by xiaocai2333** [comment]
```
[2022/12/12 10:57:33.835 +00:00] [INFO] [querynode/partition.go:74] ["add a segment to replica"] [collectionID=437728473907200005] [partitionID=437728473907200049] [segmentID=437728473907405434] [segmentType=Sealed]
[2022/12/12 11:00:36.345 +00:00] [INFO] [querynode/impl.go:501] ["loadSegmentsTask WaitToFinish done"] [collectionID=437728473907200005] [segmentIDs="[437728473907405434]"] [nodeID=10]
```

please take a lool. @bigsheeper

**Comment by xiaocai2333** [comment]
```
[2022/12/12 10:54:32.792 +00:00] [INFO] [querynode/segment_loader.go:763] ["start read delta msg from seek position to last position"] [collectionID=437728473907200005] [channel=by-dev-rootcoord-dml_0] [seekPos="channel_name:\"by-dev-rootcoord-dml_0\" msgID:\"\\304W\\334P\\272\\037\\023\\006\" msgGroup:\"by-dev-dataNode-6-by-dev-rootcoord-dml_0_437728473907200005v0\" timestamp:437794075859222530 "] [lastMsg={}]
[2022/12/12 10:57:33.728 +00:00] [INFO] [querynode/segment_loader.go:827] ["All data has been read, there is no more data"] [collectionID=437728473907200005] [channel=by-dev-rootcoord-dml_0] [msgID="xFfcULofEwY="]

```

**Comment by xiaocai2333** [comment]
same as #21103 . It has been fixed in 2.2.1. Please retry with v2.2.1. @zyy20191

**Comment by zyy20191** [comment]
> same as #21103 . It has been fixed in 2.2.1. Please retry with v2.2.1. @zyy20191

OK, I'll try again.Many thanks.
